apollo-router/src/plugins/telemetry/config_new/attributes.rs

```diff
@@ -792,6 +832,600 @@ pub(crate) struct HttpClientAttributes {
+                HTTP_RESPONSE_STATUS_CODE,
+                AttributeValue::String(response.response.status().to_string()),
+            );
+        }
+        attrs
+    }
+
+    fn on_error(&self, _error: &BoxError) -> HashMap<Key, AttributeValue> {
+        let mut attrs = HashMap::new();
+        if let Some(true) = &self.error_type {
+            attrs.insert(Key::from_static_str("error.type"), AttributeValue::I64(500));
+        }
+
+        attrs
+    }
+}
+
+impl GetAttributes<router::Request, router::Response> for HttpServerAttributes {
+    fn on_request(&self, request: &router::Request) -> HashMap<Key, AttributeValue> {
+        let mut attrs = HashMap::new();
+        if let Some(true) = &self.http_route {
+            attrs.insert(
+                HTTP_ROUTE,
+                AttributeValue::String(request.router_request.uri().to_string()),
+            );
+        }
+        let router_uri = request.router_request.uri();
+        if let Some(true) = &self.server_address {
+            if let Some(host) = router_uri.host() {
+                attrs.insert(SERVER_ADDRESS, AttributeValue::String(host.to_string()));
+            }
+        }
+        if let Some(true) = &self.server_port {
+            if let Some(port) = router_uri.port() {
+                attrs.insert(SERVER_PORT, AttributeValue::String(port.to_string()));
+            }
+        }
+        if let Some(true) = &self.url_path {
+            attrs.insert(
+                URL_PATH,
+                AttributeValue::String(router_uri.path().to_string()),
+            );
+        }
+        if let Some(true) = &self.url_query {
+            if let Some(query) = router_uri.query() {
+                attrs.insert(URL_QUERY, AttributeValue::String(query.to_string()));
+            }
+        }
+        if let Some(true) = &self.url_scheme {
+            if let Some(scheme) = router_uri.scheme_str() {
+                attrs.insert(URL_SCHEME, AttributeValue::String(scheme.to_string()));
+            }
+        }
+
+        attrs
+    }
+
+    fn on_response(&self, _response: &router::Response) -> HashMap<Key, AttributeValue> {
+        HashMap::with_capacity(0)
+    }
+
+    fn on_error(&self, _error: &BoxError) -> HashMap<Key, AttributeValue> {
+        HashMap::with_capacity(0)
+    }
+}
+
+impl GetAttribute<supergraph::Request, supergraph::Response> for SupergraphCustomAttribute {
+    fn on_request(&self, request: &supergraph::Request) -> Option<AttributeValue> {
+        match self {
+            SupergraphCustomAttribute::OperationName {
+                operation_name,
+                default,
+                ..
+            } => {
+                let op_name = request.context.get(OPERATION_NAME).ok().flatten();
+                match operation_name {
+                    OperationName::String => {
+                        op_name.or_else(|| default.clone().map(AttributeValue::String))
+                    }
+                    OperationName::Hash => todo!(),
```

Comment:
what's the plan here?

Reply:
Good question ! Haha, I forgot about that part, @BrynCooke Do you know what kind of hash we should provide ?